src/types/window.d.ts

```diff
@@ -0,0 +1,3 @@
+export interface CustomWindow extends Window {
+    dataLayer?: any[];
+}
```

Comment:
I think it's better to declare the `window` here instead of export `CustomWindow` interface.

```ts
/* eslint-disable @typescript-eslint/no-explicit-any */
export {};

declare global {
  interface Window {
    dataLayer?: any[];
  }
}
```

Reply:
@mikrowelt